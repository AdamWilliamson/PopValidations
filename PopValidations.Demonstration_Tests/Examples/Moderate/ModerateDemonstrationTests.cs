using ApprovalTests;
using System;
using System.Threading.Tasks;
using PopValidations_Tests.TestHelpers;
using Xunit;

namespace PopValidations.Demonstration_Tests.Examples.Moderate;

public class ModerateDemonstrationTests
{
    [Fact]
    public async Task BasicValidator_Validate_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new ModerateAlbumValidator());
        var album = new ModerateAlbum(
            "Disturbed",
            "https://www.disturbed1.com/sites/g/files/g2000015236/files/inline-images/Divisive_Cover-FINAL_1.jpg",
            DateTime.MinValue,
            new()
            {
                null
            }
        );

        // Act
        var results = await runner.Validate(album);
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void AlbumValidator_Describe_ToJson()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new ModerateAlbumValidator());

        // Act
        var results = runner.Describe();
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }
}
