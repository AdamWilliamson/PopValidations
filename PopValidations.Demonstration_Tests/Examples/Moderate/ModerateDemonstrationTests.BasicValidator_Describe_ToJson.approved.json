{
    "scopeMessage": null,
    "validator": null,
    "error": null,
    "description": null,
    "children": [
        {
            "scopeMessage": null,
            "validator": null,
            "error": null,
            "description": "TrackNumber",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": null,
                    "error": "Is not equal to 1",
                    "description": "Must equal to 1",
                    "children": []
                }
            ]
        },
        {
            "scopeMessage": null,
            "validator": null,
            "error": null,
            "description": "TrackName",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": null,
                    "error": "Is Null",
                    "description": "Must not be Null",
                    "children": []
                },
                {
                    "scopeMessage": null,
                    "validator": null,
                    "error": "Is not equal to A Song Title",
                    "description": "Must equal to A Song Title",
                    "children": []
                }
            ]
        },
        {
            "scopeMessage": null,
            "validator": null,
            "error": null,
            "description": "Duration",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": null,
                    "error": "Is not equal to 2.3",
                    "description": "Must equal to 2.3",
                    "children": []
                }
            ]
        },
        {
            "scopeMessage": null,
            "validator": null,
            "error": null,
            "description": "IntegerList",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": null,
                    "error": "Is Null",
                    "description": "Must not be Null",
                    "children": []
                }
            ]
        }
    ]
}